weblab/entities/tests/test_views.py

```diff
@@ -198,6 +198,52 @@ def test_version_with_two_tags(self, client, user, helpers):
                    commit, ['tag1', 'tag2'])
 
 
+@pytest.mark.django_db
+class TestModelEntityVersionCompareView:
+    def test_shows_related_experiments(self, client, user, experiment_version):
+        exp = experiment_version.experiment
+        sha = exp.model.repo.latest_commit.hexsha
+        recipes.experiment_version.make()  # another experiment which should not be included
+
+        response = client.get(
+            '/entities/models/%d/versions/%s/compare' % (exp.model.pk, sha)
+        )
+
+        assert response.status_code == 200
+        assert response.context['comparisons'] == [(exp.protocol, [exp])]
+
+    def test_returns_404_if_commit_not_found(self, user, client):
```

Comment:
It's an anonymous test (the `user` fixture was erroneously included in this and some other tests, so I will remove it). The recipes create public entities by default. Although this could probably do with a test to check that non-public entities are not being included.